Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/hibernate/LoadTestDataServiceImpl.java

```diff
@@ -48,64 +54,182 @@
+    public boolean deleteIdentifier(Identifier identifier) {
+        return identifierDAO.delete(identifier);
+    }
+
+    @Override
+    public boolean deletePhonenumber(Phonenumber phonenumber) {
+        return phonenumberDAO.delete(phonenumber);
+    }
+
+    @Override
+    public Personname getPersonnameBy(Long personnameId) {
+        return personnameDAO.read(personnameId);
+    }
+
+    @Override
+    public Address getAddressBy(Long addressId) {
+        return addressDAO.read(addressId);
+    }
+
+    @Override
+    public Identifier getIdentifierBy(Long identifierId) {
+        return identifierDAO.read(identifierId);
+    }
+
+    @Override
+    public Phonenumber getPhonenumberBy(Long phonenumberId) {
+        return phonenumberDAO.read(phonenumberId);
+    }
+
+    @Override
+    public boolean savePersonname(Personname personname) throws LoadTestDataException {
+        boolean actionResult = false;
+        if(personname.getPatient() != null){
+            if (HelperUtil.isId(personname.getPersonnameId())) {
+                actionResult = personnameDAO.update(personname);
+            }else{
+                actionResult = personnameDAO.create(personname);
+            }
+            if (!actionResult) {
+                LOG.error("DAO fail to save personname.");
+                throw new LoadTestDataException("Personname fail to save to database.");
+            }
+        }
+        return actionResult;
+    }
+
+    @Override
+    public boolean saveAddress(Address address) throws LoadTestDataException {
+        boolean actionResult = false;
+        if (address.getPatient() != null) {
+            if (HelperUtil.isId(address.getAddressId())) {
+                actionResult = addressDAO.update(address);
+            } else {
+                actionResult = addressDAO.create(address);
+            }
+            if (!actionResult) {
+                LOG.error("DAO fail to save Address.");
+                throw new LoadTestDataException("Address fail to save to database.");
+            }
         }
+        return actionResult;
+    }
 
-        LOG.info("save-patient: Action-result: {} ", actionResult);
+    @Override
+    public boolean saveIdentifier(Identifier identifier) throws LoadTestDataException {
+        boolean actionResult = false;
+        if (identifier.getPatient() != null) {
+            if (HelperUtil.isId(identifier.getIdentifierId())) {
+                actionResult = identifierDAO.update(identifier);
+            } else {
+                actionResult = identifierDAO.create(identifier);
+            }
+            if (!actionResult) {
+                LOG.error("DAO fail to save Identifier.");
+                throw new LoadTestDataException("Identifier fail to save to database.");
+            }
+        }
         return actionResult;
     }
 
+    @Override
+    public boolean savePhonenumber(Phonenumber phonenumber) throws LoadTestDataException {
+        boolean actionResult = false;
```

Comment:
To be consistent with other DML operations in GenericDAOImpl.java, initialize the actionResult to true and change the logic accordingly. Refer GenericDAOImpl.update method.